core/trino-spi/src/main/java/io/trino/spi/type/Decimals.java

```diff
@@ -264,6 +266,16 @@ public static void checkOverflow(BigInteger value)
         }
     }
 
+    public static void checkRescaleOverflow(long value, int fromScale, int toScale)
```

Comment:
What is the contract of this method? What is `value`? Can `toScale` be larger than `MAX_SHORT_PRECISION`. Let's inline this method.

Reply:
Now that I look at it, it does not serve a larger purpose as an independent function. Done.